### PR TITLE
Works for all but ubuntu

### PR DESCRIPTION
**Pull request by knightsamar**
This fixes Jenkinsfile, build and deploy scripts and also the shell scripts.

Forked out of the head of #30 but reverting the current HEAD.

### CODE REVIEW

**Comment by knightsamar** [comment]
@ramereth and @doublej472 this is ready for merging  but I would like your +1 before I merge it just in case you have any concerns

This contains all the changes from #30 sans the ubuntu templates.
I am not sure why there is a failure of  centos-7.2-ppc64 template because I can't even find that file.

File: http/ubuntu-16.04/preseed-ppc64.cfg [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+d-i passwd/user-password-again password ubuntu
+d-i passwd/username string ubuntu
+d-i pkgsel/include string openssh-server vim
+d-i pkgsel/install-language-support boolean false
+d-i pkgsel/update-policy select none
+d-i pkgsel/upgrade select full-upgrade
+d-i time/zone string UTC
+d-i user-setup/allow-password-weak boolean true
+d-i user-setup/encrypt-home boolean false
+tasksel tasksel/first multiselect standard, server
```
Comment (doublej472) [inline http/ubuntu-16.04/preseed-ppc64.cfg]: Is this file still needed if ubuntu isn't included in this PR?

File: scripts/ubuntu/cleanup.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+apt-get -y purge popularity-contest installation-report friendly-recovery laptop-detect;
+
+apt-get -y autoremove;
+apt-get -y clean;
+
+# Remove caches
+find /var/cache -type f -exec rm -rf {} \;
+
+# delete any logs that have built up during the install
+find /var/log/ -name *.log -exec rm -f {} \;
```
Comment (doublej472) [inline scripts/ubuntu/cleanup.sh]: Another ubuntu remnant

File: scripts/ubuntu/networking.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+sed -i 's/GRUB_CMDLINE_LINUX_DEFAULT\=\"/GRUB_CMDLINE_LINUX_DEFAULT\=\"net\.ifnames\=0 /' /etc/default/grub\
+  && grub-mkconfig -o /boot/grub/grub.cfg\
+  || echo "Grub2 not found!"
+
+# Fix ethernet address for network interfaces
+sed -i 's/auto\ en.*/auto\ eth0/' /etc/network/interfaces
+sed -i 's/iface\ en.*/iface\ eth0\ inet\ dhcp/' /etc/network/interfaces
+
+# Adding a 5 sec delay to the interface up, to make the dhclient happy
+echo "pre-up sleep 5" >> /etc/network/interfaces;
```
Comment (doublej472) [inline scripts/ubuntu/networking.sh]: ^

File: scripts/ubuntu/openstack.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+datasource_list: ['OpenStack']
+datasource:
+  Ec2:
+    metadata_urls: [ 'http://169.254.169.254' ]
+    timeout: 5
+    max_wait: 10
+EOF
+
+# Remove default datasource_list
+rm -f /etc/cloud/cloud.cfg.d/90_dpkg.cfg
```
Comment (doublej472) [inline scripts/ubuntu/openstack.sh]: ^

File: scripts/ubuntu/ppc.sh [hunk 5]
```diff
@@ -0,0 +1 @@
+apt-get -y install ppc64-diag
```
Comment (doublej472) [inline scripts/ubuntu/ppc.sh]: ^

File: scripts/ubuntu/sudoers.sh [hunk 6]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+sed -i -e '/Defaults\s\+env_reset/a Defaults\texempt_group=sudo' /etc/sudoers;
+
+# Set up password-less sudo for the ubuntu user
+echo 'ubuntu ALL=(ALL) NOPASSWD:ALL' >/etc/sudoers.d/99_ubuntu;
+chmod 440 /etc/sudoers.d/99_ubuntu;
```
Comment (doublej472) [inline scripts/ubuntu/sudoers.sh]: ^

File: scripts/ubuntu/update.sh [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+# Disable periodic activities of apt
+cat <<EOF >/etc/apt/apt.conf.d/10disable-periodic;
+APT::Periodic::Enable "0";
+EOF
+
+# Upgrade all installed packages incl. kernel and kernel headers
+apt-get -y dist-upgrade;
+reboot;
+sleep 60;
```
Comment (doublej472) [inline scripts/ubuntu/update.sh]: ^

**Comment by knightsamar** [comment]
@doublej472 please re-review.

File: bin/build_image.sh [hunk 8]
```diff
@@ -55,3 +55,4 @@ else
   qemu-img convert -o compat=0.10 -O qcow2 -c ${DIR_NAME}/${IMAGE_NAME} \
     $FINAL_QCOW_FILE_NAME
 fi
+
```
Comment (doublej472) [inline bin/build_image.sh]: random extra line?

File: bin/build_image.sh [hunk 9]
```diff
@@ -43,10 +43,10 @@ fi
 if [ -e chef/${TEMPLATE_NAME}/Berksfile ] ; then
   export BERKSHELF_PATH="chef/berkshelf"
   rm -rf $BERKSHELF_PATH chef/${TEMPLATE_NAME}/cookbooks chef/${TEMPLATE_NAME}/Berksfile.lock
-  berks vendor -b chef/${TEMPLATE_NAME}/Berksfile chef/${TEMPLATE_NAME}/cookbooks
+  berks vendor --delete -b chef/${TEMPLATE_NAME}/Berksfile chef/${TEMPLATE_NAME}/cookbooks
 fi
 export PACKER_LOG=1
-packer build -force -debug $(basename $TEMPLATE)
+packer build -color=false -force -debug $(basename $TEMPLATE)
```
Comment (doublej472) [inline bin/build_image.sh]: Are we planning on keeping the `-debug` switch?

Comment (knightsamar) [reply bin/build_image.sh]: Yes. That would provide additional info necessary for any debugging. Also, Jenkins client seems to handle the requirement for pressing enter on every step when prompted, so it is not detrimental.

Ideally, I would want a single debug flag passable to a build so that debug can be turned on all step but that is a feature so I won't work on it right now.

File: bin/build_image.sh (hunk 8 above)
Comment (knightsamar) [reply bin/build_image.sh]: Will remove and commit with any other changes that you suggest. Don't want to trigger another build right now.
